# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/Http3Stream.h
# Commit: 058e2449bbac
# Full Hash: 058e2449bbac40dcaa8cf1ade7464abd3b5217b0
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2020-11-05 21:37:48
# Regressor Bug: 1674922
# File Overlap Count: 1
# Description:
#   Bug 1674922 - Cleanup ReadSegments: r=necko-reviewers,kershaw
#   
#   - Move looping while calling mTransaction->ReadSegments into Http3Stream and call mTransaction->ReadSegmentsAgain. We use to loop in Http3Session which was not easy because it is not easy to find out when to leave the loop. The original code was working, but this is a better way to do this.
#   - Remove mReadyForWriteButBlocked it is not necessary, it was used only as a double check and can only be a source of bugs.
#   - Remove mContentBytesWritten, because it is not used.
# ==============================================================================

diff -r d9b078999ebc -r 058e2449bbac netwerk/protocol/http/Http3Stream.h
--- a/netwerk/protocol/http/Http3Stream.h	Thu Nov 05 11:04:20 2020 +0000
+++ b/netwerk/protocol/http/Http3Stream.h	Thu Nov 05 11:37:39 2020 +0000
@@ -35,13 +35,10 @@
   // TODO priorities
   void TopLevelOuterContentWindowIdChanged(uint64_t windowId){};
 
-  [[nodiscard]] nsresult ReadSegments(nsAHttpSegmentReader*, uint32_t,
-                                      uint32_t*);
+  [[nodiscard]] nsresult ReadSegments(nsAHttpSegmentReader*);
   [[nodiscard]] nsresult WriteSegments(nsAHttpSegmentWriter*, uint32_t,
                                        uint32_t*);
 
-  bool RequestBlockedOnRead() const { return mRequestBlockedOnRead; }
-
   void SetQueued(bool aStatus) { mQueued = aStatus; }
   bool Queued() const { return mQueued; }
 
@@ -133,7 +130,6 @@
   RefPtr<nsAHttpTransaction> mTransaction;
   nsCString mFlatHttpRequestHeaders;
   bool mQueued;
-  bool mRequestBlockedOnRead;
   bool mDataReceived;
   bool mResetRecv;
   nsTArray<uint8_t> mFlatResponseHeaders;
@@ -153,6 +149,7 @@
   uint32_t mSendingBlockedByFlowControlCount = 0;
 
   nsresult mSocketInCondition = NS_ERROR_NOT_INITIALIZED;
+  nsresult mSocketOutCondition = NS_ERROR_NOT_INITIALIZED;
 };
 
 }  // namespace net